src/test/java/software/amazon/cloudwatchlogs/emf/model/MetricsContextTest.java

```diff
@@ -75,6 +78,59 @@ public void testSerializeMoreThen100Metrics() throws JsonProcessingException {
         }
     }
 
+    @Test
+    public void testSerializeAMetricWith101DataPoints() throws JsonProcessingException {
+        MetricsContext mc = new MetricsContext();
+        int dataPointCount = 101;
+        int expectedEventCount = 2;
+        String metricName = "metric";
+        for (int i = 0; i < dataPointCount; i++) {
+            mc.putMetric(metricName, i);
+        }
+
+        List<String> events = mc.serialize();
+        assertEquals(expectedEventCount, events.size());
+        List<MetricDefinition> allMetrics = new ArrayList<>();
+        for (String event : events) {
+            allMetrics.addAll(parseMetrics(event));
+        }
+        List<Double> expectedValues = new ArrayList<>();
+        for (int i = 0; i < Constants.MAX_DATAPOINTS_PER_METRIC; i++) {
+            expectedValues.add((double) i);
+        }
+        assertEquals(expectedValues, allMetrics.get(0).getValues());
+        assertTrue(allMetrics.get(1).getValues().equals(Arrays.asList(100.0)));
+    }
+
+    @Test
+    public void testSerializeMetricsWith101DataPoints() throws JsonProcessingException {
```

Comment:
The main purpose of this test is just to verify that splitting doesn't impact other metric data points added to the context?

Reply:
Yes.